### DISCUSSION TITLE
How to connect Phantom Wallet from Telegram mini app

### DISCUSSION THREAD

**Discussion by AndyG0822** [phantom/docs#266] [Help & Support] [answered]
### Summary

Hello, Phantom experts
Hope you are all doing well.
The reason I am posting this discussion is I am facing the issue while I am working with Phantom wallet with Telegram mini app.
Now I am working on Telegram mini app which is web3 game on Solana that requires players need to sign and send tokens to some address and transaction.
I used the @solana/wallet-adapter-base and @solana/wallet-adapter-react.
But the thing is the connecting with Phantom is working when I am running my game on web browser(I am using chrome) but it is not working when I am running the game on Telegram mini app.
I think it's because the Telegram mini app does not support the web browser extension.
On Wallet adapter modal does not find the Phantom provider.
But this happens on mobile too.
Once I open the game on mobile chrome then the connecting wallet is working but while I open the game in the telegram app then it does not find the provider.
Below I posted the image how the modal looks like.
How can I solve this issue.
Please help me to resolve this issue.
Thank you for looking my issue.
Hope I am able to solve this issue here from expert's help.
Thanks again.

### Example

https://t.me/flappysolbot?game=flappy_bird

### Steps to Reproduce

![Screenshot_20240811-182246](https://github.com/user-attachments/assets/205ba255-35b7-4018-8f6c-11ca3daa48a2)
Also this is the hosted app link on Vercel.
https://flappy-bird-solana.vercel.app/

### Is there an existing discussion for this?

- [X] I have searched the existing discussions

**Comment by adamdelphantom** [comment, marked as answer]
Hello @AndyG0822,

In order to connect to Phantom from telegram you will need to implement deeplinks in your telegram app. [Here is our documentation for deeplinks.](https://docs.phantom.app/phantom-deeplinks/deeplinks-ios-and-android)

**Reply by AndyG0822** [reply]
Thank you for your help.

**Reply by teodorus-nathaniel** [reply]
Hi are you able to integrate this deeplink to your telegram app?
I am trying it myself by following this [link](https://phantom.app/learn/blog/the-complete-guide-to-phantom-deeplinks) but with webapp instead of expo
for the start of the "connect" step, I am stuck, because I don't have the `Linking.createURL("onConnect");` and I tried using https:// url, but it doesn't even show the "connect" prompt in the phantom

**Reply by adamdelphantom** [reply]
@teodorus-nathaniel Have you reviewed the various methods available [here](https://docs.phantom.app/phantom-deeplinks/deeplinks-ios-and-android)? I'm not familiar with telegram development but I have spoken with teams who have integrated phantom with telegram using deeplinks.

**Reply by Nowy334** [reply]
@adamdelphantom but Phantom deeplinks for example connect method return multiple query params included in redirect link and telegram mini app accept only one param like startapp -> https://docs.telegram-mini-apps.com/platform/start-parameter. So how do I get the public key or errors from this connect phantom deeplink method?

**Reply by teodorus-nathaniel** [reply]
> @teodorus-nathaniel Have you reviewed the various methods available [here](https://docs.phantom.app/phantom-deeplinks/deeplinks-ios-and-android)? I'm not familiar with telegram development but I have spoken with teams who have integrated phantom with telegram using deeplinks.

hello, yes, sorry for the late reply, phantom deeplinks works good

**Reply by teodorus-nathaniel** [reply]
> @adamdelphantom but Phantom deeplinks for example connect method return multiple query params included in redirect link and telegram mini app accept only one param like startapp -> https://docs.telegram-mini-apps.com/platform/start-parameter. So how do I get the public key or errors from this connect phantom deeplink method?

for me, I created one special page that is accessible from outside of telegram, so user will be redirected there and can click login to connect based on the redirect params

after the login is done, show user a button to go back to the app

**Reply by adamdelphantom** [reply]
@Nowy334 seems like the solution @teodorus-nathaniel describes is the way to go since we cannot meet the necessary scheme of parameters required by telegram.
